ee/hogai/graph/insights/nodes.py

```diff
@@ -79,7 +79,7 @@ def run(self, state: AssistantState, config: RunnableConfig) -> PartialAssistant
 
             selected_insights = self._search_insights_iteratively(search_query or "")
 
-            formatted_content = self._format_search_results(selected_insights, search_query or "")
+            formatted_content = self._format_search_results(selected_insights, search_query or "", state)
```

Comment:
Unrelated to this PR, just noticed when checking the file.

Nitpick: `_load_all_insights` seems to load all the insights without pagination. As I remember, we have cases of clients with 10k+ insights (if I'm not mistaken). Would it make sense to add pagination there? Query time/memory/etc. concerns.